pkg/opensearch/opensearch.go

```diff
@@ -45,8 +46,122 @@ func NewOpenSearchDatasource(ctx context.Context, settings backend.DataSourceIns
 func (ds *OpenSearchDatasource) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
 	res := &backend.CheckHealthResult{}
 
+	jsonDataStr := req.PluginContext.DataSourceInstanceSettings.JSONData
+	jsonData, err := simplejson.NewJson([]byte(jsonDataStr))
+	if err != nil {
+		res.Status = backend.HealthStatusError
+		res.Message = "Failed to parse settings"
+		return res, nil
+	}
+
+	flavor := jsonData.Get("flavor").MustString("")
```

Comment:
Arguably we should do this when we initialize the datasource, but I was trying to avoid changing the logic too much